eve/io/mongo/validation.py

```diff
@@ -71,10 +71,10 @@ def _remove_unique_rules_on_fields_with_unique_index(self, schema):
         # removes the unique constraint on _id. We could use the information
         # available by app.data.driver.db[datasource].index_information() to
         # remove all unnecessary unique constraints.
-        result = copy.deepcopy(schema)
```

Comment:
@julianhille Right :) That's a good point!

What about that?
* has_unique_constraints = (schema or {}).get('_id', {}).get('unique')

Another point, why are you reassigning the schema *deepcopy*  to the schema itself ?
**Instead of**
* schema = copy.deepcopy(schema)

**Something more like**
* schema_copy_or_whatever_make_sense = copy.deepcopy(schema)

Don't you think or am I missing something?